### DISCUSSION TITLE
Feature suggestion: cascadeAllAlike

### DISCUSSION THREAD

**Discussion by rudifa** [rxhanson/Rectangle#1144] [Ideas]
The PR #762 by @dirondin added `tileAll` and `cascadeAll` commands (available only via user-configured shortcuts, see discussions in #762).

It would be nice to also have `cascadeAllAlike`, which would cascade and bring to front all windows belonging to the app whose window is currently on top (in focus), leaving all other windows' layout unchanged.

@rxhanson,  @dirondin, what do you think?

**Comment by rxhanson** [comment]
Thanks for the feedback. I don't have plans to add this in, but am fine merging it in if someone wants to make a good pull request for it.

Cascading windows of an app is in [Rectangle Pro](https://rectangleapp.com/pro) (paid), but it just cascades and doesn't bring windows frontmost.

**Comment by rudifa** [comment]
@rxhanson @dirondin Actually, some weeks ago I wanted to try to implement this.
However, after I built Rectangle, I failed to persuade my Mac to authorize my build to run.
Can I find guidance somewhere on how to achieve this? 
If needed, I can try it again and supply more details on my problem.

**Reply by rxhanson** [reply]
I think you're referring to the accessibility permissions? If so, recent versions of macOS have made it kind of unbearable without a developer id. You might be able to sign it with a personal (not paid developer account) id, but I'm not sure. I think the way it currently is, for each time you build, you would have to first remove the accessibility permissions for Rectangle, then add them when the app runs.

**Reply by rudifa** [reply]
@rxhanson Thank you for the pointers, I'll investigate.

**Comment by rudifa** [comment]
@rhanson Indeed, when I build Rectangle (without any code or build settings changes)
- (having stopped and renamed the installed Rectangle.app)
- run it under debugger
- the Authorize Rectangle dialog pops up
- I open Settings, see Rectangle enabled, I disable and enable again
- quit settings
- the menu bar icon is present, its menu still says `Not Authorized to Control…`

Where is the Rectangle log, and do you know of any other log that might clarify why?
Note: I have only the free developer account.

Thanks.

**Reply by rxhanson** [reply]
This is a macOS issue, and there are no logs that would assist us here.

First, quit Rectangle if running. In system settings, accessibility settings, you need to disable _then_ remove Rectangle. Then run the app in Xcode. Then add in System Settings when prompted.

The problem is that you might have to do that every time you build. Maybe this is fixed in the latest versions of macOS, but it was broken in this way when I had taken a look at it previously. It's a pretty ugly bug macOS side, since it effectively makes it unbearable to develop an app that requires accessibility privileges if you don't have a developer account. Good luck!

**Comment by rudifa** [comment]
@rxhanson I just posted my PR #1146 **add action cascadeActiveApp; add cascadeActiveAppWindowsOnScreen**.

Thank you for the help. 

I can now repeatedly rebuild and authorize Accessibility for my Rectangle debug build.

To get there, I modified the `DEVELOPMENT_TEAM` from "" to my `LH......NA` code and then followed the `quit--remove-from-settings--build-and-run--enable-in-settings` routine.

I keep the Settings page open; when the app starts, `Rectangle` pops up in the Accessibility list; I enable it, to be rewarded with a cheerful **Welcome to Rectangle!**.

**Comment by rudifa** [comment]
Et voilà.

![Screenshot 2023-05-06 at 20 48 49 copy](https://user-images.githubusercontent.com/172404/236641868-5ae05ddc-584d-4682-9b60-3748fc641071.jpg)
